src/client_builder.rs

```diff
@@ -61,41 +155,132 @@ where
         ClientBuilder {
             domain,
             port,
-            starttls: false,
+            mode: ConnectionMode::Auto,
+            #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+            tls_kind: TlsKind::Any,
+            #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+            no_tls_verify: false,
         }
     }
 
     /// Use [`STARTTLS`](https://tools.ietf.org/html/rfc2595) for this connection.
     #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+    #[deprecated = "Use mode(ConnectionMode::StartTls) instead"]
     pub fn starttls(&mut self) -> &mut Self {
-        self.starttls = true;
+        self.mode(ConnectionMode::StartTls)
+    }
+
+    /// Use TLS for this connection.
+    #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+    #[deprecated = "Use mode(ConnectionMode::Tls) instead"]
+    pub fn tls(&mut self) -> &mut Self {
+        self.mode(ConnectionMode::Tls)
+    }
+
+    /// Sets the Connection mode to use for this connection
+    pub fn mode(&mut self, mode: ConnectionMode) -> &mut Self {
+        self.mode = mode;
         self
     }
 
-    /// Return a new [`Client`] using a `native-tls` transport.
+    /// Sets the TLS backend to use for this connection.
+    #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+    pub fn tls_kind(&mut self, kind: TlsKind) -> &mut Self {
```

Comment:
If we're now getting rid of the helpers, I think we should have this also specifically set `mode` to be `Tls` _if_ it is currently set to `Tcp`.

Reply:
I disagree with that one as they could intent mode to be StartTls, thus we really don't know.  And with my proposal of having the default AutoTls (don't allow plain text by default) this becomes even more so unexpected behavior.